### ISSUE TITLE
Vectorz and Vectors error on clojure.core.matrix/join

### ISSUE THREAD

**Issue by si14** [mikera/core.matrix#76]
``` clojure
linprog.core> (m/join (m/transpose [[1 2] [3 4]]) (m/transpose [[5 6]]))
[[1 3] [2 4] [5] [6]]
linprog.core> (m/join (m/transpose (m/array :vectorz [[1 2] [3 4]])) (m/array :vectorz [[5 6]]))
ArrayStoreException mikera.vectorz.impl.ArraySubVector  clojure.lang.RT.seqToPassedArray (RT.java:1602)
linprog.core> (m/join (m/transpose (m/array :ndarray-double [[1 2] [3 4]])) (m/array :ndarray-double [[5 6]]))
#<NDArrayDouble [[1.0 3.0] [2.0 4.0] [5.0 6.0]]>
```

(discovered during attempts to add a column to a matrix)

**Comment by mikera** [comment]
Thanks for the report, yes that looks like a bug. I'll check this out.

**Comment by mikera** [comment]
Hi Dmitry, this seems to be already fixed in the latest vectorz-clj snapshot. Should be OK in the next release, I'll create this weekend.

**Comment by mikera** [comment]
Just released 0.17.0, should all be good
